regression test for Vertx http booster is failing

### Description
The regression test for vertx http booster is failing on Travis CI. This behavior is sporadic and usually happens with OpenShift v3.6.0., v3.6.1
Its an issue with the assertion that verifies application endpoint response. Its observed that creating application route may take some time. Before that assertion invokes the application endpoint.

Hey @hrishin,
I tried reproducing it but somehow not able to. Could you please elaborate the kind of error you're watching in logs?, I build a branch 2-3 times but build completed cleanly. You can have a look at it [here](https://travis-ci.org/rohanKanojia/fabric8-maven-plugin/jobs/325138118) and [here](https://travis-ci.org/rohanKanojia/fabric8-maven-plugin/jobs/325138124).

Although one thing I noticed is the breaking watch and repeated reconnect attempts:
```
-> [INFO] Current reconnect backoff is 1000 milliseconds (T0)
-> [INFO] F8: Build spring-boot-rest-http-crud-s2i-5 Complete
-> [INFO] Current reconnect backoff is 2000 milliseconds (T1)
-> [ERROR] Exception in reconnect
-> java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@6b4a6e4d rejected from java.util.concurrent.ScheduledThreadPoolExecutor@2e48ed40[Shutting down, pool size = 1, active threads = 1, queued tasks = 0, completed tasks = 2]
-> 	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2063)
-> 	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:830)
-> 	at java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(ScheduledThreadPoolExecutor.java:326)
-> 	at java.util.concurrent.ScheduledThreadPoolExecutor.schedule(ScheduledThreadPoolExecutor.java:533)
-> 	at java.util.concurrent.Executors$DelegatedScheduledExecutorService.schedule(Executors.java:729)
-> 	at io.fabric8.kubernetes.client.dsl.internal.WatchHTTPManager$3.run(WatchHTTPManager.java:207)
-> 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
-> 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
-> 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
-> 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
-> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
-> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
-> 	at java.lang.Thread.run(Thread.java:748)
```

I wonder if we can get rid of this reconnect backoff messages, maybe increase the reconnect timeout interval. What do you think?

--- a/rt/src/test/java/io/fabric8/maven/rt/BaseBoosterIT.java
+++ b/rt/src/test/java/io/fabric8/maven/rt/BaseBoosterIT.java
@@ -42,6 +42,7 @@ import org.jboss.shrinkwrap.resolver.api.maven.embedded.EmbeddedMaven;
 import org.json.JSONObject;
 
 import java.io.*;
+import java.net.UnknownHostException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
@@ -220,7 +221,7 @@ public class BaseBoosterIT {
      * @return
      * @throws Exception
      */
-    protected Response makeHttpRequest(HttpRequestType requestType, String hostUrl, String params) throws IOException {
+    protected Response makeHttpRequest(HttpRequestType requestType, String hostUrl, String params) throws IOException, IllegalStateException {
         OkHttpClient okHttpClient = new OkHttpClient();
         MediaType json = MediaType.parse("application/json; charset=utf-8");
         params = (params == null ? new JSONObject().toString() : params);
@@ -244,12 +245,18 @@ public class BaseBoosterIT {
                 logger.info("No valid Http request type specified, using GET instread.");
                 request = new Request.Builder().url(hostUrl).get().build();
         }
-        Response response = okHttpClient.newCall(request).execute();
-        if (logger.isLoggable(Level.INFO)) {
-            logger.info(String.format("[%s] %s %s", requestType.name(), hostUrl, HttpStatus.getCode(response.code())));
-        }
 
-        return response;
+        // Sometimes nip.io is not up, so handling that case too.
+        try {
+            Response response = okHttpClient.newCall(request).execute();
+            if (logger.isLoggable(Level.INFO)) {
+                logger.info(String.format("[%s] %s %s", requestType.name(), hostUrl, HttpStatus.getCode(response.code())));
+            }
+
+            return response;
+        } catch (UnknownHostException unknownHostException) {
+            throw new IllegalStateException("No Host with name " + hostUrl + "found, maybe nip.io is down!");
+        }
     }
 
     protected int exec(String command) throws IOException, InterruptedException {

--- a/rt/src/test/java/io/fabric8/maven/rt/VertxHttpBoosterIT.java
+++ b/rt/src/test/java/io/fabric8/maven/rt/VertxHttpBoosterIT.java
@@ -18,11 +18,15 @@ package io.fabric8.maven.rt;
 
 import io.fabric8.openshift.api.model.Route;
 import okhttp3.Response;
-import okhttp3.ResponseBody;
+import org.apache.http.HttpStatus;
 import org.eclipse.jgit.lib.Repository;
+import org.json.JSONException;
 import org.json.JSONObject;
 import org.junit.After;
 import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
 
 import static io.fabric8.kubernetes.assertions.Assertions.assertThat;
 
@@ -84,9 +88,21 @@ public class VertxHttpBoosterIT extends BaseBoosterIT {
     private void assertThatWeServeAsExpected(Route applicationRoute) throws Exception {
         String hostUrl = "http://" + applicationRoute.getSpec().getHost() + testEndpoint;
 
-        Response readResponse = makeHttpRequest(HttpRequestType.GET, hostUrl, null);
+        int nTries = 0;
+        Response readResponse = null;
+        do {
+            readResponse = makeHttpRequest(HttpRequestType.GET, hostUrl, null);
+            nTries++;
+            TimeUnit.SECONDS.sleep(10);
+        } while(nTries < 3 && readResponse != null && readResponse.code() != HttpStatus.SC_OK);
+
         String responseContent = readResponse.body().string();
-        assert new JSONObject(responseContent).getString("content").equals("Hello, World!");
+        try {
+            assert new JSONObject(responseContent).getString("content").equals("Hello, World!");
+        } catch (JSONException jsonException) {
+            logger.log(Level.SEVERE, "Unexpected response, expecting json. Actual : " + responseContent);
+            logger.log(Level.SEVERE, jsonException.getMessage(), jsonException);
+        }
     }
 
     @After

